northbound/sensorthings/filter/src/main/java/org/eclipse/sensinact/northbound/filters/sensorthings/antlr/impl/paths/HistoricalLocationPathHandler.java

```diff
@@ -42,6 +42,16 @@ public Object handle(final String path) {
                 // Provider
                 return provider.getName();
 
+            case "time":
```

Comment:
Does this apply to more cases than HistoricalLocation? If so then the fix should be an extra case in 

https://github.com/gg-dim/org.eclipse.sensinact.gateway/blob/7ae2d01b92ac43c7ff38e5092c15b60d1f0b6215/northbound/sensorthings/filter/src/main/java/org/eclipse/sensinact/northbound/filters/sensorthings/antlr/impl/paths/PathUtils.java#L68

If not then you should be using `PathUtils` to find the `ResourceSnapshot` for `admin/location` not just the first resource in the list.